src/lTerm_read_line.ml

```diff
@@ -255,6 +284,14 @@ class virtual ['a] engine ?(history = []) ?(clipboard = LTerm_edit.clipboard) ?(
   let completion_index = S.map          (fun c -> c.index) completion_state in
   let history, set_history = S.create (history, []) in
   let message, set_message = S.create None in
+  let history_prefix, set_history_prefix =
+    let ev, send = E.create () in
+    let edit_changes = Zed_edit.changes edit in
+    let edit_changes = E.map (fun _ -> Zed_rope.to_string @@ Zed_edit.text edit) edit_changes in
```

Comment:
we should keep `prefix` as rope here, otherwise we'll do the conversion to a string all the time even though we only need to do it when navigating through the history